ingressPort.tex

```diff
@@ -597,16 +596,33 @@ \section{Data Trace Interface requirements}\label{sec:DataInterfaceRequirements}
 store) using \textbf{itype} 0 except for the final load or store,
 which must retire using the natural \textbf{itype} for the instruction
 (for example, a popret instruction must use \textbf{itype} 13 for the
-final load to signal the return).
+final load to signal the return).  The instruction address reported will be the same for
+each occurrence.  
+
+For example, if instructions are retired in blocks, and assuming the hart can perform one load or store per cycle,
+the following retirement sequences could be reported when pushing and popping 4 registers from the stack:
+

```

Comment:
Its not clear why "one load or one store per cycle" is material to this description. Also "retired in blocks" is not clear. Further, the `cm.push/cm.pop` are not required to be atomic. For example, a `cm.pop` that  pop's 4 data values may fault after performing 2 loads. So these load/store may be redone if the instruction was restarted following the trap.

Reply:
It's an example intended to illustrate the required behaviour using a typical hart-encoder interface.  The hart to encoder interface supports retiring instructions individually or in blocks (multiple retirements indicated by number of half-words retired), and also allows multiple blocks per cycle.  I think this is clear to someone more familiar with this spec.

Regarding faulting, that's understood.  The next paragraph discusses that so I'm unclear what you're point is here.

Reply:
For the purpose of this example, what difference did it make if the instructions retired in blocks and hart could perform one load or store per cycle. Would the trace out put not be identical i.e. 4 loads or 4 stores?
The point about faulting was that the example states that "if the instructions retired" - it is not required for `cm.pop` to retire for the loads in the `cm.pop` to be observed for data trace.

Reply:
The trace output is the same regardless.  This is not about the trace output - it's about how the hart communicates information to the encoder.  Push/pop are anomalous in that multiple loads/stores are initiated by a single instruction. 
 As described in the paragraph above, the push/pop effectively have to be "unrolled" on the interface, as an instruction retirement must be indicated with each load or store.   What I was attempting to do here was provide a worked example of this generalized behaviour (this was explicitly requested by the TG member who asked for this update).  Depending on the hart's capabilities, the configuration of the interface will vary, and I'm just describing a particular configuration in order to frame the example.  I wanted to use retirement blocks rather than single instruction retirement to illustrate how the push/pop can be retired as part of a block.  If the hart can retire multiple loads/stores per cycle then the interface is extended so more information can be communicated in parallel, but that would make the examples more involved.

In the lists, where it says "single push" or "single popret" that is intended to indicate that the hart reports one instruction retired with the address or the push or popret.  I can try to make that clearer.

Reply:
Thanks. Yes, that would make it clearer.